### PR TITLE
Add scope to products

### PR DESCRIPTION
**Pull request by nomad-alt**
### PT link: https://www.pivotaltracker.com/story/show/152671885 ###
- Add scope to products
- Add specs
- Add ingredients model

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,18 +10,35 @@
...
 
+  create_table "ingredients", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.boolean "status", default: true
+    t.integer "price"
+    t.string "name"
+  end
+
+  create_table "product_lines", force: :cascade do |t|
```
Comment (olofvi) [inline db/schema.rb]: we need to address the underscore in the serializer the front end doesn't work with it :(

File: db/schema.rb [hunk 2]
```diff
@@ -10,18 +10,35 @@
...
+
   create_table "products", force: :cascade do |t|
     t.string "name"
     t.integer "price"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.text "description"
     t.string "image"
+    t.string "of_type", default: "dish"
+    t.boolean "status", default: true
```
Comment (tochman) [inline db/schema.rb]: I'm not sure what this attribute is going to be used for. Please explain

File: spec/models/product_line_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe ProductLine, type: :model do
```
Comment (tochman) [inline spec/models/product_line_spec.rb]: I would name this ProductInridient but if you are not adding any attributes to this table there is no need for having this model at all as far as I'm concerned. Rails takes care of joint tables for you

Comment (tochman) [reply spec/models/product_line_spec.rb]: But perhaps I'm missing something. What is this model used for?

Comment (magnus-thor) [reply spec/models/product_line_spec.rb]: Joint table between product and ingredient. All the guides we read about has many to has many associaton said we needed a joint table

File: db/schema.rb (hunk 2 above)
Comment (magnus-thor) [reply db/schema.rb]: Status is used for availability. Probs should have called it available though

Comment (tochman) [reply db/schema.rb]: I think that's a good idea.

File: spec/models/product_line_spec.rb (hunk 3 above)
Comment (tochman) [reply spec/models/product_line_spec.rb]: Okay, I see that. I would suggest calling this model `ProductIngredient`

Comment (diraulo) [reply spec/models/product_line_spec.rb]: You could probably make use of a `has_and_belongs_to_many` relationship between `Product` and `Ingredient` and rails would take care of the joint table for you... Check this link for more details
http://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many

Comment (magnus-thor) [reply spec/models/product_line_spec.rb]: ok, will take a look at it.

**Review by tochman — APPROVED** [review]
Good work

**Pull request merged**